src/models/kv_cache.cpp

```diff
@@ -166,10 +166,16 @@ KV_Cache::KV_Cache(State& state)
     }
   }
 
+  auto kv_cache_size_bytes = SizeOf(type_) * shape_[0] * shape_[1] * shape_[2] * shape_[3];
   for (int i = 0; i < layer_count_ * 2; ++i) {
     presents_.push_back(
         sb_kv_caches_.empty() ? OrtValue::CreateTensor(*model_.allocator_kvcache_, shape_, type_)
                               : sb_kv_caches_[i]->CreateTensorOnStaticBuffer(shape_, type_));
+#if USE_CUDA
+    cudaMemsetAsync(presents_.back()->GetTensorMutableRawData(), 0, kv_cache_size_bytes, model_.cuda_stream_);
+#else
+    memset(presents_.back()->GetTensorMutableRawData(), 0, kv_cache_size_bytes);
+#endif
```

Comment:
Could we leverage the device interface to do this? @RyanUnderhill

Reply:
Yeah, I'll just change this when I'm removing the other calls like this. Since I probably need to add more to the Device class to do it cleanly.